internal/deb/extract_test.go

```diff
@@ -352,6 +352,40 @@ var extractTests = []extractTest{{
 		},
 	},
 	error: `cannot extract from package "test-package": path /dir/ requested twice with diverging mode: 0777 != 0000`,
+}, {
+	summary: "Hard link must be created if specified in the tarball",
+	pkgdata: testutil.MustMakeDeb([]testutil.TarEntry{
+		testutil.Dir(0755, "./"),
+		testutil.Reg(0644, "./file1.txt", "text for file1.txt"),
+		testutil.Hln(0644, "./file2.txt", "./file1.txt"),
+	}),
+	options: deb.ExtractOptions{
+		Extract: map[string][]deb.ExtractInfo{
+			"/*.txt": []deb.ExtractInfo{{
+				Path: "/*.txt",
+			}},
+		},
+	},
+	result: map[string]string{
+		"/file1.txt": "file 0644 e926a7fb",
+		"/file2.txt": "file 0644 e926a7fb",
+	},
+	notCreated: []string{},
+}, {
+	summary: "Dangling hard link must raise an error",
```

Comment:
Normally we prefer "Cannot..." without saying directly that it must raise an error because you can see that in the test case itself. What about "Cannot extract dangling hard link"?

Reply:
I think just "Dangling hard link" should be enough (if looking at the other test summaries above).